test/transforms/onehot.jl

```diff
@@ -4,28 +4,31 @@
   c = categorical([3, 2, 2, 1, 1, 3])
   t = Table(; a, b, c)
 
-  T = OneHot(1)
+  T = OneHot(1, false)
   n, c = apply(T, t)
   @test Tables.columnnames(n) == (:a_false, :a_true, :b, :c)
-  @test n.a_false == Bool[1, 0, 0, 1, 0, 0]
-  @test n.a_true  == Bool[0, 1, 1, 0, 1, 1]
+  @test n.a_false == categorical(Bool[1, 0, 0, 1, 0, 0])
+  @test n.a_true  == categorical(Bool[0, 1, 1, 0, 1, 1])
+  @test typeof(n.a_false) != typeof(categorical(Bool[1, 0, 0, 1, 0, 0]))
```

Comment:
We don't need to change the existing tests. We just need to pass the correct option to preserve old results.

Reply:
The existing tests aren't able to distinguish between the categorical columns and bool columns. So I added some modifications so that the tests are able to distinguish.

Reply:
The change we are proposing is such that we can simply pass categorical=false to get all tests passing without change. What I am saying is that you need to add new tests instead of modifying existing ones.